src/components/BookList.js

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Book from "./Book";
+
+class BookList extends React.Component {
+  static propTypes = {
+    books: PropTypes.array.isRequired,
+    filters: PropTypes.object.isRequired
+  };
+
+  applyFilters = books => {
+    const filters = this.props.filters;
+    for (let filter in filters) {
+      if (filters[filter]) {
+        books = books.filter(book => book[filter]);
+      }
+    }
+    return books;
+  };
+
+  render() {
+    let books = this.props.books;
+    books = this.applyFilters(books);
```

Comment:
`this.applyFilters(this.props.books)` would be okay I think. I don't like overwriting variables like this, but I don't mind chaining. It's possible that this should be a different object which re-renders itself if someone calls `booksCollection.applyFilter(filter)`.